packages/grid/_modules_/grid/models/colDef/gridNumericOperators.ts

```diff
@@ -109,4 +110,20 @@ export const getGridNumericColumnOperators = (): GridFilterOperator[] => [
       };
     },
   },
+  {
+    label: 'is any of',
+    value: 'isAnyOf',
+    isArrayValue: true,
+    getApplyFilterFn: (filterItem: GridFilterItem) => {
+      if (filterItem.value == null || filterItem.value.length === 0) {
```

Comment:
Checking if it's an array seems to be more reliable.

```suggestion
      if (!Array.isArray(filterItem.value) || filterItem.value.length === 0) {
```